app/src/main/java/com/vero/tasky/core/di/NetworkModule.kt

```diff
@@ -0,0 +1,43 @@
+package com.vero.tasky.core.di
+
+import com.vero.tasky.BuildConfig
+import com.vero.tasky.core.interceptors.ApiKeyHeaderInterceptor
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object NetworkModule {
+
+    @Provides
+    @Singleton
+    fun provideRetrofit(
+        okHttpClient: OkHttpClient
+    ) : Retrofit {
+        return Retrofit.Builder()
+            .baseUrl(BuildConfig.BASE_URL)
+            .addConverterFactory(MoshiConverterFactory.create())
+            .client(okHttpClient)
+            .build()
+    }
+
+    @Provides
+    @Singleton
+    fun provideOkHttpClient(
+        apiKeyHeaderInterceptor: ApiKeyHeaderInterceptor
+    ) : OkHttpClient{
+        return OkHttpClient.Builder()
+            .addInterceptor(apiKeyHeaderInterceptor)
```

Comment:
You can create the ApiKeyHeaderInterceptor directly here

.addInterceptor(ApiKeyHeaderInterceptor())

This will help you remove all the extra stuff in the interceptor that's not required (check next comment)